### DISCUSSION TITLE
formState.isValid is false after reset is called with values provided

### DISCUSSION THREAD

**Discussion by salmanbabri** [react-hook-form/react-hook-form#2281] [Q&A] [answered]
I want the submit button to be disabled if the form is in an invalid state. However, the button remains disabled until I clear a field & type again. Below is a code snippet for producing this issue. I am quite new to react, any idea what could be wrong here?

```
function TestForm() {
    const {register, handleSubmit, formState, reset} = useForm({mode: "onChange"});
    const {isValid} = formState;

    useEffect(() => {
        const data = {name: "John", count: 10};
        reset(data);
    }, [reset]);

    return (
        <div>
            <h1>Form state valid: {isValid.toString()}</h1>
            <form onSubmit={handleSubmit(d => console.log(d))}>
                <input type="text" name="name"
                       ref={register({required: true})}/>

                <input type="text" name="count"
                       ref={register({required: true})}/>

                <input type="submit" disabled={!isValid}/>
            </form>
        </div>
    );
}
```
![image](https://user-images.githubusercontent.com/15045608/87881229-56ef4e80-ca11-11ea-9830-d8cd83ae4b8f.png)

**Comment by bluebill1049** [comment, marked as answer]
provide a codesandbox plz

**Reply by salmanbabri** [reply]
https://codesandbox.io/s/elated-darwin-spjqj?file=/src/App.js

**Reply by bluebill1049** [reply]
patching; https://github.com/react-hook-form/react-hook-form/issues/2298

**Reply by bluebill1049** [reply]
hey @salmanbabri this is due to validate inputs are async, you will need to fix this with `setTimeout`, it will naturally work for async reset tho.

**Reply by salmanbabri** [reply]
@bluebill1049 Tried putting reset call inside setTimeout callback, it works for now. 👍
